coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java

```diff
@@ -279,6 +287,23 @@ public RexNode visitCall(RexCall call) {
       return Optional.empty();
     }
 
+    private Optional<RexNode> visitCast(RexCall call) {
+      final SqlOperator op = call.getOperator();
+      if (op.getKind() != SqlKind.CAST) {
+        return Optional.empty();
+      }
+
+      RexNode leftOperand = call.getOperands().get(0);
+
+      if (call.getType().getSqlTypeName().equals(SqlTypeName.DECIMAL)
+          && leftOperand.getType().getSqlTypeName().equals(SqlTypeName.TIMESTAMP)) {
```

Comment:
To be aligned with the previous syntax, could we modify it to be:
```
call.getType().getSqlTypeName() == DECIMAL && leftOperand.getType().getSqlTypeName() == TIMESTAMP
```
?

Reply:
Fixed in revision